src/core/application/useCases/pedido/index.ts

```diff
@@ -0,0 +1,31 @@
+import { IPedidoUseCase } from "./pedido.interface";
+import { Pedido } from "@domain/entities/pedido";
+import { AssertionConcern } from "@domain/base/assertionConcern";
+import { IPedidoRepository } from "@domain/repositories/pedidoRepository.interface";
+
+export class PedidoUseCase implements IPedidoUseCase {
+    private readonly pedidoRepository: IPedidoRepository;
+
+    constructor(produtoRepository: IPedidoRepository) {
+        this.pedidoRepository = produtoRepository;
+    }
```

Comment:
non-blocking:
- quando estamos fazendo só a injeção de dependencia podemos optar por um shorthand
```suggestion
    constructor(private readonly pedidoRepository: IPedidoRepository) {}
```